### PR TITLE
Don't cache failed getMesonTasks() result.

### PR DESCRIPTION
**Pull request by lukester1975**
I don't think it should rely on `changeHandler` being invoked to clear the cache.

### CODE REVIEW

**Comment by tristan957** [comment]
Could you explain the problem this fixes?

File: src/extension.ts [hunk 1]
```diff
@@ -68,11 +68,15 @@ export async function activate(ctx: vscode.ExtensionContext) {
   controller.createRunProfile("Meson run test", vscode.TestRunProfileKind.Run, (request, token) => testRunHandler(controller, request, token), true)
   ctx.subscriptions.push(controller);
 
-  let mesonTasks: Thenable<vscode.Task[]> | null = null;
+  let mesonTasks: Promise<vscode.Task[]> | null = null;
```
Comment (xclaesse) [inline src/extension.ts:71]: OOC what does that do?

**Comment by xclaesse** [comment]
@lukester1975 if it fails, what hope so you have that it will succeed next time? I think only a reconfigure has a chance to fix the error potentially.

**Comment by lukester1975** [comment]
Yeah, this is mainly about UX. Caching a failed `getMesonTasks` means further attempts to run a task do nothing, no `showErrorMessage` dialog etc.

But also reasoning about whether `changeHandler` is going to fire in every case that might "fix" the failure is too difficult (impossible?). Simpler to just not cache on failure.

Thanks

File: src/extension.ts (hunk 1 above)
Comment (lukester1975) [reply src/extension.ts:71]: There's no `catch()` on a `Thenable`.

**Comment by tristan957** [comment]
@xclaesse should we just merge this and spin a release?

**Comment by xclaesse** [comment]
@tristan957 TBH I was not convinced about this PR, if getting tasks failed once, it won't magically succeed next time. The only thing this could achieve is slowing down everything (load tasks is slow) and spamming user with the error message. Unless I missed something?

AFAIK, the only chance we have to have loading tasks succeed again is after a reconfigure, in that case we do that already.

**Comment by lukester1975** [comment]
Maybe I was unclear so let me try again: it's not about magically succeeding next time. It's about the lack of any feedback running tasks after caching the failure.

Not caching the failed result was the smallest mod I could see to achieve this. Such feedback is not spam... spam is unsolicited.

Worrying about "slowing down everything" isn't an issue for the super rare case of `getMesonTasks` failing.

**Comment by tristan957** [comment]
Seems pretty compelling to me after that explanation.

**Comment by xclaesse** [comment]
I should give it a try to see how it goes in practice. I'm in phone ATM, can't test until Monday.

The reason I'm prudent here is because when I implemented that cache it was because vscode calls that method a lot to populate menus, and parsing interpretation JSON files is super slow. So I would like to be sure it shows error only when user really does an action and not spuriously just because vscode refresh a menu.

Maybe it should cache the error state to not retry parsing and show error immediately?
